### PR TITLE
RavenDB-21342 Unary match optimization didn't handle the case of null…

### PR DESCRIPTION
**Pull request by ayende**
…s properly

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21342 

### Additional description

Unary optimization didn't handle nulls properly

### CODE REVIEW

File: src/Corax/Queries/MultiUnaryMatch.cs [hunk 1]
```diff
@@ -20,6 +20,26 @@ public unsafe struct MultiUnaryItem
      */
     public FieldMetadata Binding;
     public DataType Type;
+
+    public string RightAsString()
+    {
+        return Type switch
```
Comment (arekpalinski) [inline src/Corax/Queries/MultiUnaryMatch.cs:26]: Check the warning

File: src/Corax/Queries/MultiUnaryMatch.cs [hunk 2]
```diff
@@ -20,6 +20,26 @@ public unsafe struct MultiUnaryItem
...
+        {
+            DataType.Double => DoubleValueRight.ToString(CultureInfo.InvariantCulture),
+            DataType.Long => LongValueRight.ToString(CultureInfo.InvariantCulture),
+            DataType.Slice => SliceValueRight.ToString()
+        };
+    }
+    
+    public string LeftAsString()
+    {
+        return Type switch
```
Comment (arekpalinski) [inline src/Corax/Queries/MultiUnaryMatch.cs:37]: Warning

**Comment by redknightlois** [comment]
LGTM

File: src/Corax/Queries/MultiUnaryMatch.cs (hunk 2 above)
Comment (ayende) [reply src/Corax/Queries/MultiUnaryMatch.cs:37]: fixed

File: src/Corax/Queries/MultiUnaryMatch.cs (hunk 1 above)
Comment (ayende) [reply src/Corax/Queries/MultiUnaryMatch.cs:26]: fixed

File: src/Corax/Queries/MultiUnaryMatch.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Corax/Queries/MultiUnaryMatch.cs:37]: Did you forget to push?

Comment (ayende) [reply src/Corax/Queries/MultiUnaryMatch.cs:37]: Sorry, yes

**Pull request merged**